### PR TITLE
Css composition rule

### PR DESCRIPTION
**Pull request by ahuth**
@juanca, can you take a look at this when you get a chance, please? This adds a rule preventing string interpolation for classNames.

I expect that backfilling will need to be done in Mavenlink.

Also note that version numbers don't need to be bumped, as our publishing process will take care of that.

### CODE REVIEW

**Comment by ahuth** [comment]
@brandonduff, would love to have your thoughts as well, if you have time.

**Comment by brandonduff** [comment]
So here's a use case of this in Mavenlink:

https://github.com/mavenlink/mavenlink/blob/7395164e3228ba3dc6aae51b3d6038e308068244/frontend/features/side-panel/components/custom-fields/edit-styling.jsx#L4

Basically, it adds on disabled/enabled styling to a component. Kind of unusual, and there might be a better pattern for this.

I suppose we can always ignore lint rules, so one example probably shouldn't be blocking. Since CSS composition is almost always what you want, I think the rule makes sense. The best test will probably be to cut a bigmaven branch that implements this and see what breaks. Maybe we'll find a bunch of legit uses of string concatenation; maybe not. I would bet this rule would be helpful.

**Comment by ahuth** [comment]
@brandonduff, cases like that **_might_** (or might not) be able to be used with css composition.

If not, it might be a one off, as you noted. Thanks for the example, though.

Also as you noted, the _real_ test will be to see how much work it'll take to get Mavenlink passing. Planning on doing that and reporting back here.

Thanks for your insights.

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     "packages/*"
   ],
   "scripts": {
+    "publish-canary": "lerna publish --canary --npm-client=npm --registry=https://registry.npmjs.org/",
```
Comment (ahuth) [inline package.json]: This command publishes alpha versions of changed packages linked to specific commits, and without the autogenerated tags or changes to the package.json.

Useful for testing in Mavenlink.

**Comment by ahuth** [comment]
Turns out there's 30 places in Mavenlink violating this.

**Comment by juanca** [comment]
Are those legit violations? I think if we can keep it to a minimum, we can ignore the rule for those lines?

**Comment by ahuth** [comment]
That's exactly what we did in https://github.com/mavenlink/mavenlink/pull/11102.

Many of those violations aren't "legit", in that the solution could've been solved initially with composition. However, the work to fix them now was too great and I disabled'em.

**Comment by ahuth** [comment]
Thanks!

**Pull request merged**